cub/test/catch2_test_device_transform.cu

```diff
@@ -618,3 +622,44 @@ C2H_TEST("DeviceTransform::Transform aligned_base_ptr", "[device][device_transfo
   STATIC_REQUIRE(::cuda::std::is_constructible_v<kernel_arg>);
   STATIC_REQUIRE(::cuda::std::is_copy_constructible_v<kernel_arg>);
 }
+
+namespace cudax = cuda::experimental;
+
+C2H_TEST("DeviceTransform::Transform cudax::async_device_buffer", "[device][device_transform]", algorithms)
```

Comment:
ditto we should move to cudax

Reply:
I kind of disagree. This test is more about `cub::DeviceTransform` correctly working on the `async_device_vector`, than whether the vector can be transformed.

Reply:
Moved the example to cudax, because cudax tests already have CUB available.